### PR TITLE
Networking V2: support external network resource

### PR DESCRIPTION
**Pull request by ozerovandrei**
Add "router:external" OpenStack Networking service attribute to the resource_openstack_networking_network_v2.
Add tests for create and update of network with that attribute.

For #356

### CODE REVIEW

File: openstack/resource_openstack_networking_network_v2.go [hunk 1]
```diff
@@ -53,6 +54,12 @@ func resourceNetworkingNetworkV2() *schema.Resource {
 				ForceNew: false,
 				Computed: true,
 			},
+			"external": &schema.Schema{
+				Type:     schema.TypeString,
```
Comment (jtopjian) [inline openstack/resource_openstack_networking_network_v2.go]: We can use `TypeBool` here to automatically check the true/false values.

File: website/docs/r/networking_network_v2.html.markdown [hunk 2]
```diff
@@ -75,6 +75,10 @@ The following arguments are supported:
     by any tenant or not. Changing this updates the sharing capabalities of the
     existing network.
 
+* `external` - (Optional)  Specifies whether the network resource has the
+    external routing facility. Changing this updates the external attribute of
+    the existing network.
```
Comment (jtopjian) [inline website/docs/r/networking_network_v2.html.markdown]: "Valid values are `true` and `false`. Defaults to `false`."

File: openstack/resource_openstack_networking_network_v2.go (hunk 1 above)
Comment (ozerovandrei) [reply openstack/resource_openstack_networking_network_v2.go]: @jtopjian I didn't think about it when I copied checks from the `shared` attribute. Thank you, I'll fix it.

**Comment by ozerovandrei** [comment]
@jtopjian I actually thought that I must ask you to edit Neutron's policy.json to allow those actions. But `testAccPreCheckAdminOnly(t)` looks good too. I'll add it to tests then.

**Comment by jtopjian** [comment]
@ozerovandrei The downside is that those test aren't run in CI at this time, but that will most likely be enabled in the future.

**Comment by ozerovandrei** [comment]
@jtopjian is there something else that needs to be fixed?

**Comment by jtopjian** [comment]
@ozerovandrei I apologize, I haven't had time to do another review on this. I should be able to sometime today.

**Comment by ozerovandrei** [comment]
@jtopjian no problem, I just thought maybe I missed something 😄

**Comment by HassanIstanbouli** [comment]
Thanks @ozerovandrei  for your interest in this issue .
i hope this issue solved in the next Terraform release because currently i have to create the external network manually form openstack dashboard .
P.S : i saw a commit ( form you i guess ! ) in "terraform-provider-openstack" repository related to this issue but after a few hours it's gone !!! .
so what's happening now ?!!

**Comment by ozerovandrei** [comment]
@HassanIstanbouli Hello. There is no need to worry, those changes will be added eventually.

**Comment by HassanIstanbouli** [comment]
ok @ozerovandrei 
but by the way i would like to refer to another issue i faced with terraform : 
when you want to add an interface to a router you can't specify an optional IP to this interface ,it's automatically take the gateway IP you already specify when you create the subnet ,although you can specify it in openstack dashboard .
this issue cause an error if you want to link 2 router interfaces to the same subnet .
should i post a new issue related to this problem ?!

**Comment by ozerovandrei** [comment]
@HassanIstanbouli its better to discuss that in a separate issue of course.

But you can create `networking_router_interface_v2` resources and specify both subnet and port inside them: [documentation](https://www.terraform.io/docs/providers/openstack/r/networking_router_interface_v2.html).
Attaching several ports from a single subnet to a single router sounds very strange to me (but I'm not a Neutron expert, so maybe this schema will work).
I guess that you should check this schema via the standard `openstack` client and if it works, you can try `networking_router_interface_v2` Terraform resource.

**Comment by HassanIstanbouli** [comment]
@ozerovandrei  i told you , i tried that from openstack dashboard and i could specify an IP to it .
i need two interfaces connected to the same subnet because i want to isolate some instances in different network .
briefly my scenario is : 
i have 1 external network  and 2 internal network , the first internal have 2 instances and it connects to the external by a router ,the second internal have 2 instances and it connects to the first internal by another router . so the problem is that the second router can't attach second interface from second internal because the gateway for this second internal has already taken form first router interface . so the solution is to specify another IP (optional) , terraform doesn't have attribute to specify this optional IP .
35973170_1816751655057230_6223367398125207552_n

**Pull request merged**